### ISSUE TITLE
Incorporate new score from MolePro into Ranker?

### ISSUE THREAD

**Issue by amykglen** [RTXteam/RTX#1320]
noticed that MolePro now provides a 'CMAP similarity' score for some of the edges returned in this query: https://arax.ncats.io/?r=5628

![Screen Shot 2021-03-23 at 8 47 53 PM](https://user-images.githubusercontent.com/49423686/112251827-2ce1d200-8c19-11eb-9c9b-0f14a30fb72a.png)

should we plug this into the Ranker so it knows what to do with it? (looks like one of the edge attributes links to further information on what this score is? https://clue.io/cmap)

assigning @finnagin, but feel free to reassign however makes sense, of course.. :)

**Comment by finnagin** [comment]
It looks like this outputs percentages (0-100) but from [this](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5952941/) it looks like cmap provides ranks for pairs with rank being 1 the strongest up-regulation while rank n (where n is the size of the dataset) is the strongest down-regulation. Still trying to figure out how they get from this to percentages and how up vs down are represented. Is 0% strong down-regulation or would both strong down-regulation and strong up-regulation be represented as 100%?

**Comment by finnagin** [comment]
This was a still do but is now implemented. Though I'm not sure if the best way to handle this is to divide by 100 or put through some sort of logarithmic function. I've done a few different MolePro queries and I don't seem to see any results below 90 coming back. Should we try to expand the difference between 99 and 90?

**Comment by finnagin** [comment]
Yeah Amy's query is old TRAPI so it won't work unfortunately. [Here is where it is implemented in the code.](https://github.com/RTXteam/RTX/blob/83a5138d7f62fe4226018966934c750459f18002/code/ARAX/ARAXQuery/ARAX_ranker.py#L499-L501)

**Comment by dkoslicki** [comment]
Great, thanks!

**Comment by chunyuma** [comment]
Thanks @finnagin and @dkoslicki, I'm asking @amykglen to figure out how to get those scores directly from `MolePro` api instead of calling DSL to get those scores. I will follow Finn's method to process those values and scale them between -100 to 100.

**Comment by chunyuma** [comment]
@finnagin ah, I see, thanks for clarifying this. So using your scaling method actually ignores the up-regulation and down-regulation information, right? The scaled value between 0 to 1 can only reflect the "strong" or "weak" connection, right?

**Comment by finnagin** [comment]
Yeah, that's correct.

**Comment by finnagin** [comment]
Closing as this looks to be implemented.
